Author name issue resolution

ekongobie opened the pull request:
Solving issue #65

ekongobie commented:
@techytushar Thanks for the review. 
I have removed the migration file from the commit.

ekongobie commented:
@techytushar @rtgdk 
I have updated the PR with the changes you requested. Please could you review?

rtgdk commented:
Also you need to change the license_request/ page as well to include the author name field.

ekongobie commented:
@rtgdk Ok. I have started working on both of your requested changes, expect updates ASAP. Thanks.

ekongobie commented:
@rtgdk @techytushar 
Should I also add the License author name in the generated xml?

ekongobie commented:
@rtgdk @techytushar 
I have updated the branch, please could you review?
Thanks.

rtgdk commented:
@ekongobie Looks good to me. Can you post a screenshot of test passing successfully? (Only those Test Cases that you changed)

rtgdk commented:
@goneall Please verify if this looks good. You might want to run `makemigrations` and `migrate` commands before running the server for this branch. Also since it include database change we might want to provide default value to `author name` for previous license requests in production.

goneall commented on src/app/models.py:
```diff
@@ -28,6 +28,7 @@ class LicenseNames(models.Model):
     name = models.CharField(max_length=200)
 
 class LicenseRequest(models.Model):
+    licenseAuthorName = models.CharField(max_length=100)
```
Please add a default of "" (blank)

ekongobie replied:
@goneall I have updated the PR.
```suggestion
    licenseAuthorName = models.CharField(max_length=100, default="")
```

rtgdk commented:
@ekongobie Great! Can you also add some test verifying that default feature works and some other edge cases. I'll merge the PR as soon as this is done

rtgdk commented:
@ekongobie  You could write a new test if the Name of the Author is not given, django should take the default one and there should be no error on the JavaScript form check on the frontend. Since you mentioned `SubmitNewLicenseRequest` test, it would be great if you could write those tests down and any other you see fit.

ekongobie commented:
@rtgdk I would like to leave out the tests for the submitNewLicense for now, then when this PR is closed, I shall open another one for the tests on the submitNewLicense.

I have completed the test for the default author name. For that, I needed to login via github in the test client, then perform the license submission test.

However, I needed to use my github authentication info to accomplish this. This is why I have not yet updated this PR.

If that is fine by you, I would like to create an spdx github account which shall be used only for the tests.
The github keys in the secret.py file are not enough to accomplish this.
Here is the information I shall need from a github account:
`{
        "uid": "<github user ID>",
        "access_token": "<token>",
        "token_type": "bearer",
        "auth_time": <authentication time>,
        "login": "<github user name>",
        }`

So, If I am allowed to create an spdx github user for tests, let me know, alongside with the public authentication creds.

goneall commented:
I noticed the travis build is failing with the following error:

```django.db.utils.OperationalError: no such column: app_licenserequest.licenseAuthorName```

Is there something we need to do to migrate the schema for the tests to pass?

goneall commented:
> If that is fine by you, I would like to create an spdx github account which shall be used only for the tests.

As long as we aren't violating the github terms of service, we can create a test github account for testing.  I did a quick look to see if I could add one specific to SPDX and it looks like we just need to create a separate user.

goneall commented:
I tested this out locally and it looks good.  If we could resolve the unit test failure and rebase to the latest, I can merge in the changes.

ekongobie commented:
@goneall 
I have updated the travis.yml file to make and execute the django migrations before running tests. That was the cause of the travis build failure.
Now, the build **passes**.

I have also added the [methods](https://github.com/spdx/spdx-online-tools/pull/77/commits/ca548551b5e031aa9b3f41c88eafc31bf3b1cc11#diff-ddc103bd7786c61ce583326ae268203e)
- getAccessToken():
- getGithubUserId():
- getGithubUserName():
That will store variables required for github authentication necessary for the submit license tests to be executed.
By keeping the return values of those methods to `None` and using the `skipIf` [decorator](https://github.com/spdx/spdx-online-tools/pull/77/commits/ca548551b5e031aa9b3f41c88eafc31bf3b1cc11#diff-6819ab36c12ce7e40d6600a21b6fc9b6), I can guarantee that the submit license tests will only be executed if those values are filled.

goneall commented:
@ekongobie Thanks for the changes.  I noticed that the license submittal tests add issues to the license-list-XML repository which is used by the legal team to track the actual issue.

I'm thinking we should add a test repository and use that URL for the issues.

We will need to have a way for the test to specify the test repository - perhaps an optional parameter with the repository URL.

I created a test repository (a clone of the original LicenseListXML) at https://github.com/spdx/TEST-LicenseList-XML

ekongobie commented:
@goneall 
I have done the necessary changes to create issues in the test-repo during tests.
But I don't have the necessary API data as is the case with the license-list repo.

Here is what I mean:
![mk1](https://user-images.githubusercontent.com/45539225/53784262-fc854900-3f14-11e9-8a04-8177ae535c26.png)
![mk3](https://user-images.githubusercontent.com/45539225/53784371-5259f100-3f15-11e9-832c-e7a7e6927998.png)


I think there is something extra which you need to activate on the repo for the API to function normally.

goneall commented:
@ekongobie thanks for updating the unit tests.  I added you as a collaborator with write access to the https://github.com/spdx/TEST-LicenseList-XML repo.

Let me know if that works.

ekongobie commented:
@goneall @techytushar 
Sorry, I just found out that;
Instead of `https://api.github.com/spdx/TEST-LicenseList-XML/issues`
I had to make requests to` https://api.github.com/repos/spdx/TEST-LicenseList-XML/issues`

It now works.
I am now preparing my PR update.

Thanks.

goneall commented:
> It now works.

Excellent - I'll be on a plane tomorrow, but I should be able to merge the PR on Thursday or Friday.

@ekongobie Thanks for working on this!

ekongobie commented:
@goneall @rtgdk @techytushar 
I have updated the PR.

rtgdk commented:
@ekongobie Can you please update this branch with master? I updated the verbosity of the travis test. I want to see a more verbose successful travis run of this PR.

ekongobie commented:
@rtgdk 
Thanks for pointing that out.
I just resolved conflicts with the master branch and committed my changes.
You can now see the successful verbose travis run of this PR.

goneall commented:
@rtgdk I'm traveling today and tomorrow - feel free to merge these changes into master if the changes look good to you.  I'll deploy the changes over the weekend.

rtgdk commented:
@ekongobie The test `test_post_submit` is getting skipped in travis because of empty github parameters. Can you post a screenshot of the successful run of this test with the your github token and api parameters? I just want to confirm the test works fine without getting skipped.
We would have to create a test user for setting up these github api parameters later.

ekongobie commented:
@rtgdk Here are images of the test passing.
![t5](https://user-images.githubusercontent.com/45539225/54071989-d218f000-4274-11e9-9c81-5bcc1a38f480.png)
![t6](https://user-images.githubusercontent.com/45539225/54071990-d2b18680-4274-11e9-910c-3f993099d90f.png)

rtgdk commented:
@ekongobie  Thanks! Nice work!
